ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -419,34 +429,68 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
    */
   private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, AdminRequest adminRequest)
       throws StoreException, IOException {
-    boolean hostHealthy = false; //used to determine if the host is ever unhealthy
+    ServerHealthStatus serverHealthStatus = ServerHealthStatus.BAD; //used to determine if the server is ever unhealthy
+    JSONArray brokenDisks = new JSONArray();        // indicate which disks didn't pass the disk healthcheck
+    JSONArray unstablePartitions = new JSONArray(); // indicate which partitions aren't in leader/standby state
 
     if (this.storeManager instanceof StorageManager) {
       StorageManager storageManager = (StorageManager) this.storeManager;
-      hostHealthy = true;
+      serverHealthStatus = ServerHealthStatus.GOOD;
```

Comment:
Can this.storeManager be an instance of something else ?
Also, I find the assignment to this variable a bit redundant BAD -> GOOD -> BAD.
One way around this is to eliminate L438
and add a snippet at the end of this loop

if (unstablePartitions.length()) then status = BAD.

Reply:
storeManager can be CloudStorageManager which doesn't have the resources we need
But as for the the redundant logic. I see your point and just added a ternary operation now. If we can't perform a disk healthcheck then we wouldn't know the state of the disk so it's considered bad